### PR TITLE
happy thoughts project

### PR DESCRIPTION
**Pull request by lunek1**
## Netlify link
https://celebrated-shortbread-0673ce.netlify.app/

## Collaborators
Individual project

### CODE REVIEW

File: src/components/List/ThoughtList.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  /*Converts to minutes after 60 seconds*/
+  useEffect(() => {
+    fetchThoughts();
+    const fetchInterval = setInterval(fetchThoughts, 60000);
+    return () => {
+      clearInterval(fetchInterval);
+    };
+  }, []);
+
+  const calculateTimeDifference = (timestamp) => {
```
Comment (El1an3) [inline src/components/List/ThoughtList.jsx:36]: Would have been nice to have days, month etc. too as mentioned in your readme.

File: src/components/List/ThoughtList.jsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    if (timeDifferenceInSeconds < 60) {
+      return `${timeDifferenceInSeconds} seconds ago`;
+    } else {
+      const timeDifferenceInMinutes = Math.floor(timeDifferenceInSeconds / 60);
+      return `${timeDifferenceInMinutes} minutes ago`;
+    }
+  };
+
+  /*Function for the scroll to the top-button*/
```
Comment (El1an3) [inline src/components/List/ThoughtList.jsx:51]: I like that, didn't think of it myself. 👍

File: README.md [hunk 3]
```diff
@@ -1,35 +1,21 @@
...
+- Convert 60min+ to show h + min.
+- Change the like function to maybe just add 1 like. Now you can click endlessly. Add appropriate styling so that it's clear for the user which thoughts they have liked.
+- When the user submits a thought it doesn't show up straight away unless you refresh. Would have wanted to change that and added an animation to it.
+- Designed a cuter header! Putting more effort, time and making it look more professional. Now I mainly focused on the basic requirements.
+- Add more features!
 
-### View it live
-
-Every project should be deployed somewhere. Be sure to include the link to the deployed project so that the viewer can click around and see what it's all about.
+🚨 There seem to be some kind of css issue when I try the netlify link on my phone. On my computers it looks fine. The arrow svg inside my scroll to the top button isn't centered within the button. The submit button font is white instead of black. Let me know if you also get this or if there is any other issue. Will look into it!
```
Comment (El1an3) [inline README.md:17]: I get the same on my phone (white font and arrow not centered).

File: src/components/Input/ThoughtInput.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        placeholder="Add your thought..."
+        value={inputValue}
+        onChange={handleChange}
+        className="input-field"
+      />
+      <div className="input-container">
+        <button type="submit" className="submit-button">
+          ❤️ Send Happy Thought ❤️
+        </button>
+        <div className={inputValue.length > 140 ? "counter red" : "counter"}>
```
Comment (El1an3) [inline src/components/Input/ThoughtInput.jsx:57]: I don't get a red counter here. It only gives me -xy characters. The styling for "red" seems to be missing.

**Review by El1an3 — COMMENTED** [review]
Good job Cornelia! You can be proud. The app meets all the basic requirements. I like the custom header and that you where able to add the counter. However, the part where the counter should turn red is not working. I think you only styled "counter" but not "red". I also like the scroll to bottom button.

**Review by dzc1 — APPROVED** [review]
Hello Cornelia,

Fantastic job on your project! I'm pleased to inform you that you've successfully met all the criteria required for a passing grade. Your work in the `ThoughtList.jsx` file, particularly with logic stands out impressively, save for one minor hiccup. You might have noticed that when adding a new thought in your app, it necessitates a manual refresh to display. To circumvent this in your `ThoughtList.jsx`, you can simply adjust your useEffect dependency array from:

 ```
/*Fetches thoughts and sets up a fetch interval to update every 60 seconds.*/
  /*Converts to minutes after 60 seconds*/
  useEffect(() => {
    fetchThoughts();
    const fetchInterval = setInterval(fetchThoughts, 60000);
    return () => {
      clearInterval(fetchInterval);
    };
  }, []);
```


to: 

 ```
/*Fetches thoughts and sets up a fetch interval to update every 60 seconds.*/
  /*Converts to minutes after 60 seconds*/
  useEffect(() => {
    fetchThoughts();
    const fetchInterval = setInterval(fetchThoughts, 60000);
    return () => {
      clearInterval(fetchInterval);
    };
  }, [thoughts]);
```
This should do the job !! 


keep up the great work 💪💪💪💪

**Pull request closed without merging**